public/app/plugins/datasource/prometheus/configuration/PromSettings.tsx

```diff
@@ -171,346 +172,353 @@ export const PromSettings = (props: Props) => {
               <InlineField
-                label="Cache level"
+                label="Incremental querying (beta)"
                 labelWidth={PROM_CONFIG_LABEL_WIDTH}
                 tooltip={
                   <>
-                    Sets the browser caching level for editor queries. Higher cache settings are recommended for high
-                    cardinality data sources.
+                    This feature will change the default behavior of relative queries to always request fresh data from
+                    the prometheus instance, instead query results will be cached, and only new records are requested.
+                    Turn this on to decrease database and network load.
                   </>
                 }
                 interactive={true}
+                className={styles.switchField}
                 disabled={options.readOnly}
               >
-                <Select
-                  width={40}
-                  onChange={onChangeHandler('cacheLevel', options, onOptionsChange)}
-                  options={cacheValueOptions}
-                  value={
-                    cacheValueOptions.find((o) => o.value === options.jsonData.cacheLevel) ?? PrometheusCacheLevel.Low
-                  }
+                <Switch
+                  value={options.jsonData.incrementalQuerying ?? false}
+                  onChange={onUpdateDatasourceJsonDataOptionChecked(props, 'incrementalQuerying')}
                 />
               </InlineField>
             </div>
           </div>
-        )}
 
-        <div className="gf-form-inline">
-          <div className="gf-form max-width-30">
-            <InlineField
-              label="Incremental querying (beta)"
-              labelWidth={PROM_CONFIG_LABEL_WIDTH}
-              tooltip={
+          <div className="gf-form-inline">
+            {options.jsonData.incrementalQuerying && (
+              <InlineField
+                label="Query overlap window"
+                labelWidth={PROM_CONFIG_LABEL_WIDTH}
+                tooltip={
+                  <>
+                    Set a duration like 10m or 120s or 0s. Default of 10 minutes. This duration will be added to the
+                    duration of each incremental request.
+                  </>
+                }
+                interactive={true}
+                disabled={options.readOnly}
+              >
                 <>
-                  This feature will change the default behavior of relative queries to always request fresh data from
-                  the prometheus instance, instead query results will be cached, and only new records are requested.
-                  Turn this on to decrease database and network load.
+                  <Input
+                    onBlur={(e) =>
+                      updateValidDuration({ ...validDuration, incrementalQueryOverlapWindow: e.currentTarget.value })
+                    }
+                    className="width-25"
```

Comment:
This input is too big. Could you please adjust that?
![image](https://github.com/grafana/grafana/assets/820946/325d78b5-fdbc-464f-9d99-eb81613cd003)

Reply:
I just read your comment here https://raintank-corp.slack.com/archives/C03FZ9T50R1/p1690324905204519 So feel free to ignore this